TYPO3.Flow/Classes/TYPO3/Flow/Aop/Builder/ProxyClassBuilder.php

```diff
@@ -282,12 +282,20 @@ protected function getProxyableClasses(array $classNamesByPackage)
         $proxyableClasses = array();
         foreach ($classNamesByPackage as $classNames) {
             foreach ($classNames as $className) {
-                if (!in_array(substr($className, 0, 15), $this->blacklistedSubPackages)) {
-                    if (!$this->reflectionService->isClassAnnotatedWith($className, \TYPO3\Flow\Annotations\Aspect::class) &&
-                        !$this->reflectionService->isClassFinal($className)) {
-                        $proxyableClasses[] = $className;
+                if (in_array(substr($className, 0, 15), $this->blacklistedSubPackages)) {
+                    continue;
+                }
+                if ($this->reflectionService->isClassAnnotatedWith($className, Flow\Aspect::class)) {
+                    continue;
+                }
+                if ($this->reflectionService->isClassFinal($className)) {
+                    /** @var Flow\Proxy $proxyAnnotation */
+                    $proxyAnnotation = $this->reflectionService->getClassAnnotation($className, Flow\Proxy::class);
```

Comment:
I just restructured this a bit in order to avoid even more nesting, but this is the new behavior:
Previously we skipped if `isClassFinal()` was true, now only if it's true and the class is not whitelisted via `proxy` annotation.